tests/test_piiremover.py

```diff
@@ -43,12 +43,37 @@ def test_pii_remover_catches_phone_numbers(self):
         self.assertTrue(pii_remover('This is not a real telephone number (03069) 990123!') == success)
         self.assertTrue(pii_remover('This is not a real telephone number 03069 990123!') == success)
 
+    def test_pii_remover_catches_credit_card_numbers(self): 
+            
+        '''
+        Source for fake card numbers: 
+        https://www.paypalobjects.com/en_US/vhelp/paypalmanager_help/credit_card_numbers.htm
+        '''
+
+        success = 'My fake credit card number is {{CREDIT CARD}}, thank you!'
+        
+        self.assertTrue(pii_remover('My fake credit card number is 378282246310005, thank you!') == success)           
```

Comment:
BTW Rather than:
```
    self.assertTrue(pii_remover('My fake credit card number is 378282246310005, thank you!') == success)
AssertionError: False is not true
```
I suggest:
```
    self.assertEqual(pii_remover('My fake credit card number is 378282246310005, thank you!'), success)
AssertionError: 'My fake credit card number is {{PASSPORT NUMBER}}10005, thank you!' != 'My fake credit card number is {{CREDIT CARD}}, thank you!'
- My fake credit card number is {{PASSPORT NUMBER}}10005, thank you!
?                                 ^ ---- ^^^^^^^^  -----
+ My fake credit card number is {{CREDIT CARD}}, thank you!
?                                 ^^^^^^^^  ^
```
because of the better output when it fails.

You could keep your specific matchers for each type and add a more general matcher for long numbers. This would avoid the partial match problem and identify other numbers like vehicle VINs or pension numbers or all sorts of things. It makes sense to me that long numbers are likely to be identifiable, whatever they are. (Only exception is when they quote a well-known number, like a phone number for an office that they called). And your regex could allow spaces & dashes in the number too. I would have thought we can live with a few false positives, but less so false negatives. You backed away from this though, so maybe you have good reason.

You'd have to deal with the case when multiple matchers match a string. In this example, clearly a longer match is preferable. But if they are the same length you could go by a certain ordering. I think scrubadub may do the former, because of the sort, there were no tests for this.

And if the number is redacted by the wrong matcher, but fully redacted, then that's still ok. It's fine to have a test showing the redaction was a bit funny. The test is useful for documenting this funny behaviour - you could add a comment saying it's not perfect.

Regarding an admin page to change the redaction, yes if its simple and clear exactly what you're going to do. I imagine it's going to be you doing the changes, so if you're at all unsure about the process, I'd be tempted to suggest you wait until you're trialing it, do the process manually (flask or psql command-line shell).

I imagine a further tempting step would be to create a UI to allow more general rules - wildcards or new regexes, but best avoid that. You'd need an interface to see how it affects the rate of true/false positives, and I think you'd be better off doing this away from the website - better in a jupyter notebook or similar.